seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-common/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/AbstractJdbcIT.java

```diff
@@ -95,14 +96,38 @@ public abstract class AbstractJdbcIT extends TestSuiteBase implements TestResour
 
     abstract GenericContainer<?> initContainer();
 
+    protected URLClassLoader getUrlClassLoader() throws MalformedURLException {
+        if (urlClassLoader == null) {
+            urlClassLoader =
+                    new URLClassLoader(
+                            new URL[] {new URL(driverUrl())},
+                            AbstractJdbcIT.class.getClassLoader());
+            Thread.currentThread().setContextClassLoader(urlClassLoader);
+        }
+        return urlClassLoader;
+    }
+
+    protected Class<?> loadDriverClassFromUrl() {
+        try {
+            return getUrlClassLoader().loadClass(jdbcCase.getDriverClass());
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    "Failed to load driver class: " + jdbcCase.getDriverClass(), e);
+        }
+    }
+
+    protected Class<?> loadDriverClass() {
+        try {
+            return Class.forName(jdbcCase.getDriverClass());
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    "Failed to load driver class: " + jdbcCase.getDriverClass(), e);
+        }
+    }
+
     protected void initializeJdbcConnection(String jdbcUrl)
-            throws SQLException, ClassNotFoundException, MalformedURLException,
-                    InstantiationException, IllegalAccessException {
-        URLClassLoader urlClassLoader =
-                new URLClassLoader(
-                        new URL[] {new URL(driverUrl())}, AbstractJdbcIT.class.getClassLoader());
-        Thread.currentThread().setContextClassLoader(urlClassLoader);
-        Driver driver = (Driver) urlClassLoader.loadClass(jdbcCase.getDriverClass()).newInstance();
+            throws SQLException, InstantiationException, IllegalAccessException {
+        Driver driver = (Driver) loadDriverClass().newInstance();
```

Comment:
quick question: what's different between change before and after?

Reply:
Before: every time the `initializeJdbcConnection` method is called, it will load the driver class from url
After: most drivers ard loaded from local class path, and only gbase driver is loaded from url.

Reply:
Got it. Shall we remove these useless `driverUrl` implement? And add default implement for `driverUrl` method.

Reply:
> Got it. Shall we remove these useless `driverUrl` implement? And add default implement for `driverUrl` method.

The driverUrl is still required by the engine containers (used in extendedFactory).

I find there are some references to `URLClassLoader` in it cases of selectdb and doris, but these classed are disabled now, maybe we can make some improvements when the community decide to enable them.

Reply:
oh, I see.